python/pyspark/sql/connect/column.py

```diff
@@ -30,8 +30,8 @@
 
 def _bin_op(
     name: str, doc: str = "binary function", reverse: bool = False
-) -> Callable[["ColumnRef", Any], "Expression"]:
-    def _(self: "ColumnRef", other: Any) -> "Expression":
+) -> Callable[["Column", Any], "Expression"]:
+    def _(self: "Column", other: Any) -> "Expression":
```

Comment:
I think this is a bit unclear. For example,  Spark Connect's `Expression` has binary operators like `pyspark.sql.Column` does. If Spark Connect's `ColumnRef` indicates a column reference only, it should be something like `AttributeReference`. cc @cloud-fan

Reply:
yeah I was stopped several times when implementing API by current expressions in Python side. My current focus is to improve here.....

Reply:
This Python `column` is to match the Scala `Column` I believe: https://github.com/apache/spark/blob/865a3ded2ea1ca86be93df58205882bc509b98cd/sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala#L151

Reply:
`Expression` is basically a wrapper for `Column`

Reply:
cc @grundprinzip too FYI

Reply:
I am sure the current `Column` (or `ColumnRef`) is a concept of  reference attribute based on its `to_plan` impl:

```
    def to_plan(self, session: Optional["RemoteSparkSession"]) -> proto.Expression:
        """Returns the Proto representation of the expression."""
        expr = proto.Expression()
        expr.unresolved_attribute.unparsed_identifier = self._unparsed_identifier
        return expr
```

Reply:
Class comment says the same:

```
class ColumnRef(Expression):
    """Represents a column reference. There is no guarantee that this column
    actually exists. In the context of this project, we refer by its name and
    treat it as an unresolved attribute. Attributes that have the same fully
    qualified name are identical"""
 ```

Reply:
Ok I guess my question is do we want to match what PySpark is doing? PySpark pretty much uses `Column` as `Expression`, which is exposed on API surface.

Connect is exposing this `ColumnRef` to the API surface but only meaning for unresolved attribute reference. 

My initial thought for this PR is to at least matching what PySpark is doing. Rename first, then expand this class to match PySpark.